sherpa/astro/ui/tests/test_astro_ui_unit.py

```diff
@@ -3168,6 +3168,117 @@ def test_set_bkg_not_a_background(clean_astro_ui):
         ui.set_bkg(ui.Data1D("x", [1, 2], [1, 2]))
 
 
+@pytest.mark.parametrize("id", [None, 1, "x"])
+@pytest.mark.parametrize("bkg_id", [None, 1, "y"])
+def test_delete_bkg_no_data(id, bkg_id, clean_astro_ui):
+    """Check we can delete a background when nothing exists.
+
+    This just checks the code does not error out.
+    """
+
+    ui.delete_bkg(id, bkg_id)
+
+
+@pytest.mark.parametrize("id", [None, 1, "x"])
+@pytest.mark.parametrize("bkg_id", [None, 1, "y"])
+def test_delete_bkg_no_bkg(id, bkg_id, clean_astro_ui):
+    """Check we can delete a background when nothing exists.
```

Comment:
I'm not happy with this behaviour. Do we need it for backwards compatibility? I'm worried about silent errors, e.g. I have several datasets and I want to remove the background for one of them. I mistype the id, and don't get an error, so I assume it all worked, when in fact it didn't do anything. 
I would prefer to raise an error when a user tries to remove a background that does not exist.

Reply:
This mirrors `delete_data`:

```
>>> from sherpa.astro import ui
WARNING: imaging routines will not be available, 
failed to import sherpa.image.ds9_backend due to 
'RuntimeErr: DS9Win unusable: Could not find ds9 on your PATH' 
>>> ui.delete_data()
>>> ui.delete_data(1)
>>> ui.delete_data("x")
```